test/test_utils/security.py

```diff
@@ -187,3 +262,280 @@ def get_ecr_scan_allowlist_path(image_uri):
+    upgraded_image_tag = f"{original_image_repository}-{original_image_tag}-upgraded"
+    new_image_uri = f"{registry}/{new_repository_name}:{upgraded_image_tag}"
+    return new_image_uri
+
+
+def run_upgrade_on_image_and_push(image, new_image_uri):
+    """
+    Creates a container for the image being tested. Runs apt update and upgrade on the container
+    and the commits the container as new_image_uri. This new image is then pushed to the ECR. 
+
+    :param image: str
+    :param new_image_uri: str
+    """
+    ctx = Context()
+    docker_run_cmd = f"docker run -id --entrypoint='/bin/bash' {image}"
+    container_id = ctx.run(f"{docker_run_cmd}", hide=True, warn=True).stdout.strip()
+    apt_command = "apt-get update && apt-get upgrade"
+    docker_exec_cmd = f"docker exec -i {container_id}"
+    attempt_count = 0
+    apt_ran_successfully_flag = False
+    # When a command or application is updating the system or installing a new software, it locks the dpkg file (Debian package manager).
+    # Since we have multiple processes running for the tests, there are cases when one of the process locks the dpkg file
+    # In this scenario, we get error: ‘E: Could not get lock /var/lib/dpkg/lock’ while running apt-get update
+    # That is why we need multiple tries to ensure that it succeeds in one of the tries.
+    # More info: https://itsfoss.com/could-not-get-lock-error/
+    while True:
+        run_output = ctx.run(f"{docker_exec_cmd} {apt_command}", hide=True, warn=True)
+        attempt_count += 1
+        if not run_output.ok:
+            test_utils.LOGGER.info(
+                f"Attempt no. {attempt_count} on image: {image}"
+                f"Could not run apt update and upgrade. \n"
+                f"Stdout is {run_output.stdout} \n"
+                f"Stderr is {run_output.stderr} \n"
+                f"Failed status is {run_output.exited}"
+            )
+            sleep(2 * 60)
+        elif run_output.ok:
+            apt_ran_successfully_flag = True
+            break
+        if attempt_count == 10:
+            break
+    if apt_ran_successfully_flag == False:
+        raise ValueError(
+            f"Could not run apt update and upgrade on image: {image}. \n"
+            f"Stdout is {run_output.stdout} \n"
+            f"Stderr is {run_output.stderr} \n"
+            f"Failed status is {run_output.exited}"
+        )
+    ctx.run(f"docker commit {container_id} {new_image_uri}", hide=True, warn=True)
+    ctx.run(f"docker rm -f {container_id}", hide=True, warn=True)
+    ctx.run(f"docker push {new_image_uri}", hide=True, warn=True)
+
+
+def _invoke_lambda(function_name, payload_dict={}):
+    """
+    Asyncronously Invokes the passed lambda.
+
+    :param function_name: str, name of the lambda function
+    :param payload_dict: dict, payload to be sent to the lambda
+    """
+    lambda_client = boto3.client("lambda", region_name=os.getenv("REGION"))
```

Comment:
Please change this to:
```suggestion
    lambda_client = boto3.client("lambda", region_name=test_utils.DEFAULT_REGION)
```
or whatever region the Lambda Function is meant to be deployed to.

The Lambda Function is not replicated across regions, right? There needs to be only one Lambda Function for this, and you can just set the region to be that.

In a hypothetical scenario, if the workflow were being executed in `us-east-1`, for some reason, we don't want it to look for the Lambda Function in a region that it doesn't exist.

Reply:
Great suggestion. Made changes, thanks!!